includes/diagnostics/class-wc-stripe-diagnostics-recorder.php

```
@@ -0,0 +1,400 @@
+		$this->promoter    = $promoter ?? new WC_Stripe_Diagnostics_Outcome_Promoter( $store );
+		$this->snapshotter = $snapshotter ?? new WC_Stripe_Diagnostics_Order_Snapshotter( $store, $redactor );
+	}
+
+	/**
+	 * Wire up the filter subscriptions. Called once during plugin init.
+	 *
+	 * Gated on the diagnostics toggle: when the merchant has the feature
+	 * disabled, no hooks register and the recorder is effectively dormant.
+	 */
+	public function init(): void {
+		if ( class_exists( 'WC_REST_Stripe_Diagnostics_Controller' ) && ! WC_REST_Stripe_Diagnostics_Controller::is_enabled() ) {
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'WC_REST_Stripe_Diagnostics_Controller' in method 'init'. (undefined)

(StaticAccess)